byterun/caml/weak.h

```diff
@@ -18,10 +18,122 @@
 #ifndef CAML_WEAK_H
 #define CAML_WEAK_H
 
-#ifdef CAML_INTERNALS
-
```

Comment:
The header should probably defend against inclusion in a C++, just like other header files:
```c
#ifdef __cplusplus
extern "C" {
#endif

...

#ifdef __cplusplus
}
#endif
```